integrations/gcp/gcp_core/overrides.py

```diff
@@ -17,7 +17,24 @@ class GCPCloudResourceConfig(ResourceConfig):
     selector: GCPCloudResourceSelector
 
 
+class GCPResourceSelector(Selector):
+    preserve_api_response_case_style: bool | None = Field(
+        default=None,
+        alias="preserveApiResponseCaseStyle",
+        description=(
+            "Controls whether to preserve the gcloud asset API's original field format instead of using protobuf's default snake_case. "
+            "When False (default): Uses protobuf's default snake_case format (existing behavior). "
+            "When True: Preserves the specific asset API's original format (e.g., camelCase for PubSub). "
+            "If not set, defaults to False to maintain existing behavior (snake_case for all APIs)."
+        ),
+    )
```

Comment:
We've agreed that since few kinds are affected, let users make specific mapping kinds. Lets leave the `cloudResource` kind since it typically uses the assets api and where the versioned_resource object that is used in the mapping is unaffected by the case style change.